dali/test/python/test_dali_stateless_operators.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import nvidia.dali as dali
+import nvidia.dali.fn as fn
+from nvidia.dali.pipeline import Pipeline
+from collections.abc import Iterable
+from test_utils import compare_pipelines
+from nose_utils import assert_raises
+
+# Test configuration
+batch_size = 8
+test_data_shape = [25, 15, 3]
+test_data_layout = "HWC"
+
+
+def tensor_list_to_array(tensor_list):
+    if isinstance(tensor_list, dali.backend_impl.TensorListGPU):
+        tensor_list = tensor_list.as_cpu()
+    return tensor_list.as_array()
+
+
+# Check whether a given pipeline is stateless
+def check_is_pipeline_stateless(pipeline_factory, iterations=10):
+    pipe = pipeline_factory()
+    for _ in range(iterations):
+        pipe.run()
+    # Compare a pipeline that was already used with a fresh one
+    compare_pipelines(pipe, pipeline_factory(), batch_size, iterations)
+
+
+# Provides the same random batch each time
+class RandomBatch:
+    def __init__(self):
+        rng = np.random.default_rng(1234)
+        self.batch = [rng.integers(255, size=test_data_shape, dtype=np.uint8)
+                      for _ in range(batch_size)]
+
+    def __call__(self):
+        return self.batch
+
+
+def check_single_input(op, batch=True, exec_async=True, exec_pipelined=True, **kwargs):
```

Comment:
Do you need the `batch`, `exec_async` and `exec_pipelined` args?

Reply:
No, the args were specified in code I was using as reference, and I thought that it might be useful in future. If you think it is more elegant to remove them, I don't see any problem with that.

Reply:
Sure, I am fine either way.